src/nimgraphviz/private/graphs.nim

```diff
@@ -0,0 +1,304 @@
+import os
+import osproc
+import streams
+import strformat
+import strutils
+
+import "./edges"
+
+import tables
+export tables
+
+
+type
+  NodeCollection* = ref object of RootObj
+    ## Base type for both types of graphs.
+    ## You should not have to instanciate it directly.
+    name*: string    ## The name of the graph
+    graphAttr*: Table[string, string]  ## A table of key-value pairs
+                       ## describing the layout and
+                       ## appearence of the graph
+    nodeAttrs*: Table[string, Table[string, string]] ## A table of all the nodes and their attributes
+
+  Graph* = ref object of NodeCollection
```

Comment:
As you don't use `methods` (fortunately), there is no reason to use inheritance (`ref object of`) here or for `NodeCollection`. All procedures that take a `GenericGraph` are just that, generic. So no need for OO features for that either.

Aside from that I prefer the variant object kind, but that's personal preference of course.